app/fragments/NewAddressFormatFragment.tsx

```diff
@@ -0,0 +1,123 @@
+import { Platform, View, Text } from "react-native";
+import { fragment } from "../fragment";
+import { useNetwork, useTheme } from "../engine/hooks";
+import { useBounceableWalletFormat, useSelectedAccount } from "../engine/hooks/appstate";
+import { useSafeAreaInsets } from "react-native-safe-area-context";
+import { StatusBar } from "expo-status-bar";
+import { ScreenHeader } from "../components/ScreenHeader";
+import { t } from "../i18n/t";
+import { useTypedNavigation } from "../utils/useTypedNavigation";
+import { Typography } from "../components/styles";
+import { RoundButton } from "../components/RoundButton";
+import { useCallback, useMemo } from "react";
+import { useAddressBookContext } from "../engine/AddressBookContext";
+import { queryClient } from "../engine/clients";
+import { Queries } from "../engine/queries";
+
+export const NewAddressFormatFragment = fragment(() => {
+    const theme = useTheme();
+    const network = useNetwork();
+    const safeArea = useSafeAreaInsets();
+    const navigation = useTypedNavigation();
+    const [bounceableFormat, setBounceableFormat] = useBounceableWalletFormat();
+    const selectedAccount = useSelectedAccount();
+    const addressBook = useAddressBookContext().state;
+    const contacts = addressBook.contacts;
+
+    const { oldAddressString, newAddressString } = useMemo(() => {
+        return {
+            oldAddressString: selectedAccount?.address.toString({ testOnly: network.isTestnet }),
+            newAddressString: selectedAccount?.address.toString({ testOnly: network.isTestnet, bounceable: false })
+        }
+    }, [selectedAccount, network.isTestnet]);
+
+    const switchFormat = useCallback(async () => {
+        // Prefetch contacts contract info for current account at least
+        Object.keys(contacts).forEach((contactEntrie) => {
+            if (!queryClient.getQueryData(Queries.ContractInfo(contactEntrie[0]))) {
+                queryClient.prefetchQuery(Queries.ContractInfo(contactEntrie[0]));
```

Comment:
We prefetch so that when user goes to contacts tab contacts are displayed in new (or old, depending on selected option) correctly (instead of re-rendering individually when ContactsFragment gets presented)

Reply:
✅